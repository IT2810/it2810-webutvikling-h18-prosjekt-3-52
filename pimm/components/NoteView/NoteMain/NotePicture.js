import React from 'react';
import { Image, Button, View } from 'react-native';
import { ImagePicker, Permissions } from 'expo';

export default class NotePicture extends React.Component{
    state = {
        image: null,
        permission: false,
    };

    render() {
        let { image } = this.state;


        if (image === null) {
            return (
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <Button
                        title="Pick an image that fits to your note"
                        onPress={this.pickImage}
                        color="#B8860B"
                    />

                    <View style={{ borderColor: 'black', borderWidth: 1, backgroundColor: 'beige', width: 280, height: 300 }}>
                        
                    </View>
                </View>
            );
        } else {
            return (
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <Button
                        title="Pick an image that fits to your note"
                        onPress={this.pickImage}
                        color="#B8860B"
                    />

                    <View style={{ borderColor: 'black', borderWidth: 1, backgroundColor: 'beige', width: 280, height: 300 }}>
                        <Image source={{ uri: image }} style={{ width: 280, height: 300 }} />
                    </View>
                </View>
            );
        }


    }

    componentWillReceiveProps(nextProps){
        if (this.props.image!==nextProps.image || nextProps.fl) {
            this.setState({image: nextProps.image});
            this.props.sfl();
        }
    }


    pickImage = async () => {
        if (this.state.permission === false) {
            const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
            if(status === "granted"){
                this.setState({permission: true});
            }

        }
        if (this.state.permission === true) {
            let result = await ImagePicker.launchImageLibraryAsync({
                allowsEditing: false,
            });

            if (!result.cancelled) {
                this.setState({ image: result.uri });
                this.props.sendImageUp(result.uri);

            }
        }
    };
}