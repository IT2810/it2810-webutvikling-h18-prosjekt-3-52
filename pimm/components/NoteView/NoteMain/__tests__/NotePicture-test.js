import React from 'react';
import 'react-native';
import NotePicture from '../NotePicture.js';
import renderer from 'react-test-renderer';

test('NotePicture renders correctly', ()=>{
	const np = renderer.create(<NotePicture />).toJSON();
    expect(np).toMatchSnapshot();
    expect(np.type).toBe("View");
	expect(np.children[0].props.accessibilityComponentType).toBe("button");
	expect(np.children[1].type).toBe("View");
	expect(np.children[1].children[0].type).toBe("Image");
});

test('setting image source', ()=>{
	const np = renderer.create(<NotePicture />).getInstance();
	expect(np.state.image).toBe(null);
	np.setState({image: "/data/user/0/host.exp.exponent/cache/cropped1814158652.jpg"});
	expect(np.state.image).toBe("/data/user/0/host.exp.exponent/cache/cropped1814158652.jpg");
});