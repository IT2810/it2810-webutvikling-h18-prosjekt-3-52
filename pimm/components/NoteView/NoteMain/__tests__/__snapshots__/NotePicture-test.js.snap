// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotePicture renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Pick an image that fits to your note
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#DCDCDC",
        "borderColor": "black",
        "borderWidth": 1,
        "height": 300,
        "width": 280,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": null,
        }
      }
      style={
        Object {
          "height": 300,
          "width": 280,
        }
      }
    />
  </View>
</View>
`;
